tests/test_concurrency.py

```diff
@@ -0,0 +1,50 @@
+import itertools
+
+import pytest
+
+from anonlink import concurrency
+
+DATASET_SIZES = (0, 1, 100)
+DATASET_NUMS = (0, 1, 2, 3)
+DATASETS = tuple(itertools.chain.from_iterable(
+    itertools.product(DATASET_SIZES, repeat=n) for n in DATASET_NUMS))
+CHUNK_SIZE_AIMS = (1, 10, 100)
+
+
+
+@pytest.mark.parametrize('datasets', DATASETS)
+@pytest.mark.parametrize('chunk_size_aim', CHUNK_SIZE_AIMS)
+def test_chunk_size(datasets, chunk_size_aim):
+    # Guarantee: chunk_size_aim / 4 < chunk_size < chunk_size_aim * 4.
+    # It may be possible to prove a better bound.
+    chunks = concurrency.split_to_chunks(chunk_size_aim,
+                                         dataset_sizes=datasets)
+    for chunk in chunks:
+        size = 1
+        i0, i1 = chunk['datasetIndices']
+        for a, b in chunk['ranges']:
+            assert a <= b
+            size *= b - a
```

Comment:
General point: It's usually a good idea to check the results of the function you're testing against fixed pre-calculated values, rather than simply that the values are consistent. Here `a` and `b` might be zero and the tests would pass.  Consistency != correctness.

Reply:
That’s a good point. In this case, however, I didn’t want to make any guarantees about how a and b in each case. Rather, I preferred to make sure that the chunk sizes made sense and that every comparison was included in exactly one chunk.